### ISSUE TITLE
Initial bootstrap fails with TLS/SSL configured

### ISSUE THREAD

**Issue by ac-astuartkregor** [codership/galera#457]
Apologies if this repo is the wrong place for this issue, or if I'm doing something daft.

## Problem encountered
**Tl;dr** - TLS prevents a cluster from being created for the first time.

On a fresh install of MariaDB on CentOS 7.3, if TLS/SSL is configured, the first node of the cluster will successfully start using `galera_new_cluster`, but other nodes cannot join the cluster and fail to start.

However, if the cluster is configured __without__ TLS, other nodes can join the cluster successfully.

Once this initial bootstrap of a cluster is complete, TLS can be configured and the cluster can be successfully bootstrapped once again.

## Steps to reproduce
On a number of servers:
1. Install a CentOS 7.3 minimal image and setup networking to the internet and between the servers in the cluster
2. Install the percona-release-0.1-4 RPM for their repos
3. Install MariaDB repos
4. Install the following packages:
    * MariaDB-server-10.1.22-1
    * galera-25.3.19-1
    * percona-toolkit-3.0.1-1
    * percona-xtrabackup-24-2.4.6-2
    * socat-1.7.2.2-5
5. Generate necessary certificates and configure MariaDB, wsrep and SST to use TLS
6. Secure the installation in the normal manner
7. Start one node with `galera_new_cluster`
8. Attempt to start the mysql service on a second node

## Notes
* If TLS is not set up, then the installation works just fine and nodes can join the cluster.
* If a node has been in the cluster before and TLS is configured afterwards, this will also work.
* If a node has never been in a cluster and has TLS configured, it will not be able to join the cluster.

The galera and MariaDB packages installed are coming from the MariaDB CentOS yum repository.

**Comment by ayurchen** [comment]
Hi, yes, this may be a wrong project to report the bug, since it is about MariaDB product:
 1. without configuration it is not exactly clear what does "TLS/SSL configured" mean.
 2. without the logs it is impossible to say what and where exactly is the issue.

**Comment by ac-astuartkregor** [comment]
I will try and get logs for you however I feel the issue is related to Galera since I am trying to set up a multi-node cluster. It could be something to do with the SST provider (xtrabackup-2) or something else but it'd be interesting to hear opinions.

TLS/SSL configured means adding the `ssl-ca`, `ssl-key` and `ssl-cert` variables for mysqld and client/mysql sections using correct and verified certificates. Also setting `wsrep_provider_options` to include `socket.ssl_key`, `socket.ssl_cert`, `socket.ssl_ca` variables.

From what I recall, the problems seem to occur when setting the options in `wsrep_provider_options`.
